### DISCUSSION TITLE
Modify survey-text plugin to add time limit

### DISCUSSION THREAD

**Discussion by mayaraza** [jspsych/jsPsych#2995] [Q&A]
Hello,

This modification of the survey-text plugin was suggested in this discussion (https://github.com/jspsych/jsPsych/discussions/1146). 

// end trial if trial_duration is set
if (trial.trial_duration !== null) {
jsPsych.pluginAPI.setTimeout(function () {
// when the time limit is up, automatically submit the form and end the trial
// by simulating a next button click
display_element.querySelector('#jspsych-survey-time-form').requestSubmit();
}, trial.trial_duration);
}

I pasted this into my survey-text plugin file, but the time-limit is not working. The plugin works as normal but with no time limit, even though i set trial_duration. I must be missing something - please can you help?

Thank you,
Maya

**Comment by ChristopheBossens** [comment]
Hi,

What version of the framework are you using?

There are two places that require an adjustment in the plugin code. The first is where the parameters are defined. There you should add (after the autocomplete parameter):

```
          trial_duration: {
            type: jspsych.ParameterType.INT,
            pretty_name : "Trial duration",
            default: null,
          },
```

In the code that you show above, the id of the form is also referenced incorrectly. It should work with the following code: 

```
          // Define a timeout function:
          if(trial.trial_duration !== null){
            jsPsych.pluginAPI.setTimeout(function (){
                display_element.querySelector("#jspsych-survey-text-form").requestSubmit();
            }, trial.trial_duration)
          }
```

**Comment by JAQuent** [comment]
Thanks to your help I was able to create a custom plugin for 7.1.2 (https://gist.github.com/JAQuent/7c0161a9770bd9853eee8155e05d87bb) but I think this should really be part main function.
